File: simplified-feeds-api/src/main/java/org/nypl/simplified/feeds/api/FeedFacetPseudoTitleProviderType.java
package org.nypl.simplified.feeds.api;

/**
 * A title provider for a pseudo-facet.
 *
 * This interface essentially exists to allow pseudo-facets to take their titles
 * from Android resources without this module actually having to have a hard
 * dependency on Android.
 */

public interface FeedFacetPseudoTitleProviderType
{
  /**
   * @param t A pseudo-facet
   *
   * @return A title for the given pseudo-facet
   */

  String getTitle(
    FeedFacet.FeedFacetPseudo.FacetType t);
}

File: simplified-feeds-api/src/main/java/org/nypl/simplified/feeds/api/FeedFacetPseudoTitleProviderType.kt
package org.nypl.simplified.feeds.api

/**
 * A title provider for a pseudo-facet.
 *
 * This interface essentially exists to allow pseudo-facets to take their titles
 * from Android resources without this module actually having to have a hard
 * dependency on Android.
 */

interface FeedFacetPseudoTitleProviderType {
  val collection: String
  val collectionAll: String
  val sortBy: String
  val sortByAuthor: String
  val sortByTitle: String
}
